b2sdk/session.py

```diff
@@ -100,12 +100,7 @@ def authorize_account(self, realm, application_key_id, application_key):
         :param str application_key: user's :term:`application key`
         """
         # Clean up any previous account info if it was for a different account.
-        try:
-            old_account_id = self.account_info.get_account_id()
-            old_realm = self.account_info.get_realm()
-            if application_key_id != old_account_id or realm != old_realm:
-                self.cache.clear()
-        except MissingAccountData:
+        if not self.account_info.is_same_key(application_key_id, realm):
```

Comment:
This is better than it used to be, but still not good. It used to only work well for master key, now it works for the same key (nonmaster1 -> nonmaster1), but still it doesn't work between two non-master keys and there is actually a regression for switching between non-master and master key (the cache gets cleared in this particular case now, while it was not cleared in the past).

In order to actually fix it you need to compare old_account_id with the new account id (that we'll only find out if we manage to do the remote call).

The cache should not be cleared if we are moving between two different keys from the same account. Also, it needs more tests.

I suggest that you split this PR - ship `is_same_key()` first so that you can proceed with the CLI part, then open a new PR to solve the cache clearing issue.